packages/jira-adapter/src/filters/project.ts

```
@@ -94,6 +94,61 @@ const removeComponents = async (projectId: number, client: JiraClient): Promise<
   })))
 }
 
+const isIdResponse = createSchemeGuard<{ id: string }>(Joi.object({
+  id: Joi.string().required(),
+}).unknown(true).required(), 'Received an invalid project id response')
+
+const getProjectId = async (projectKey: string, client: JiraClient): Promise<string> => {
+  const response = await client.getSinglePage({
+    url: `/rest/api/3/project/${projectKey}`,
+  })
+
+  if (!isIdResponse(response.data)) {
+    throw new Error('Received an invalid project id response')
+  }
+
+  return response.data.id
+}
+
+const isFieldConfigurationSchemeResponse = createSchemeGuard<{
+  values: {
+    fieldConfigurationScheme: {
+      id: string
+    }
+  }[]
+}>(Joi.object({
+  values: Joi.array().items(Joi.object({
+    fieldConfigurationScheme: Joi.object({
+      id: Joi.string().required(),
+    }).unknown(true),
+  }).unknown(true)),
+}).unknown(true).required(), 'Received an invalid field configuration scheme response')
+
+const deleteFieldConfigurationScheme = async (
+  change: Change<InstanceElement>,
+  client: JiraClient,
+): Promise<void> => {
+  const instance = await resolveValues(getChangeData(change), getLookUpName)
+  const response = await client.getSinglePage({
+    url: `/rest/api/3/fieldconfigurationscheme/project?projectId=${instance.value.id}`,
+  })
+
+  if (!isFieldConfigurationSchemeResponse(response.data)) {
+    throw new Error('Received an invalid field configuration scheme response')
+  }
+
+  if (response.data.values.length === 0) {
```

Just wondered - if we are in this func, than it means that we failed to create the fieldConfigurationScheme because there is already an existing one, but if we get that there are no fieldConfigurationScheme here, we just continue. Isn't it weird? don't we won't to throw in that case?

---

It is a weird case. I believe it can happen if somehow we will get 500 when creating a none-jsm project so it won't have a fieldConfigurationScheme (only jsm project creates fieldConfigurationScheme when created). I don't think I would want to throw an error here in that case (the original error of the 500 error code will be thrown) but I will add a log here